app/services/renewal_reminder_service_base.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class RenewalReminderServiceBase < ::WasteExemptionsEngine::BaseService
+  def run
+    expiring_registrations.each do |registration|
+      begin
+        send_email(registration)
+      rescue StandardError => e
+        Airbrake.notify e, registration: registration.reference
+        Rails.logger.error "Failed to send first renewal email for registration #{registration.reference}"
+      end
+    end
+  end
+
+  private
+
+  def send_email
+    raise(NotImplementedError)
+  end
+
+  def expiring_registrations
```

Comment:
This handles the criteria for the first reminder email. The problem is the A/C for the second reminder email has this

> If the registration triggering the email appears to be already renewed via Magic link, this email won’t be triggered to be sent.

So we somehow need to cover this difference.